packages/sdk-core/src/proving/proving-manager-thread.ts

```diff
@@ -32,16 +30,43 @@ export type ProvingManagerSetupInput = {
   fee: number;
   refund: number;
   provingKey: Uint8Array;
+}
 
-  roots?: Array<Uint8Array>;
-  refreshCommitment?: string;
-};
+export type ProvingManagerSetupInput<T extends NoteProtocol> = ProvingManagerPayload[T];
 
-type PMEvents = {
-  proof: ProvingManagerSetupInput;
+type PMEvents<T extends NoteProtocol = 'mixer'> = {
+  proof: [T, ProvingManagerSetupInput<T>];
   destroy: undefined;
 };
 
+/**
+ * @param roots - Roots for anchor API
+ * @param refreshCommitment - Refresh commitment in hex representation ( without prefix `0x` ) Required for anchor, ignored for the mixer
+ * */
+export type AnchorPMSetupInput = MixerPMSetupInput & {
```

Comment:
Can we just keep them separate from the Mixer? Especially since this looks like it probably takes in other parameters we don't even need, right? Just keep them all well-defined and separate so it's clear.